app/components/channel_system_message/channel_system_message.js

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {Posts} from 'mattermost-redux/constants';
+import FormattedMarkdownText from 'app/components/formatted_markdown_text';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {ViewTypes} from 'app/constants';
+
+export default class ChannelSystemMessage extends React.PureComponent {
+    static propTypes = {
+        post: PropTypes.object.isRequired,
+        theme: PropTypes.object.isRequired,
+        navigator: PropTypes.object.isRequired,
+        textStyles: PropTypes.object.isRequired,
+    };
+
+    render() {
+        const {post, theme, navigator, textStyles} = this.props;
+
+        const style = getStyleSheet(theme);
+
+        const systemMessageRenderers = {
+            [Posts.POST_TYPES.HEADER_CHANGE]: renderHeaderChangeMessage,
+            [Posts.POST_TYPES.DISPLAYNAME_CHANGE]: renderDisplayNameChangeMessage,
+            [Posts.POST_TYPES.PURPOSE_CHANGE]: renderPurposeChangeMessage,
+            [Posts.POST_TYPES.CHANNEL_DELETED]: renderChannelDeletedMessage,
+        };
+
+        return post.type ? systemMessageRenderers[post.type](post, navigator, style, textStyles, theme) : null;
```

Comment:
Sorry, I think you must've misunderstood me here. I was meaning that we should check if `systemMessageRenderers[post.type]` is null before calling it like
```javascript
const renderer = systemMessageRenderers[post.type];
if (!renderer) {
    return null;
}

return systemMessageRenderers[post.type](post, navigator, style, textStyles, theme);
```

Reply:
I don't think that this would be a worthy change. We already narrowed down post types to matching ones and have a handler here for each type. If we go this route, we should probably test every variable for being null.

Reply:
We don't really care if `post.type` is null though since that won't cause an error. If `systemMessageRenderers[post.type]` is null, then the app will crash. It's less important to null check all of the fields since their prop types are marked with `.isRequired` which tells people using this code that they should be null, but it's harder to document that the component has to be provided a specific post type

Reply:
@hmhealey first, I did not understand what you are driving at regarding `null` here. perhaps you did not understand this piece, or maybe I am confused. if you want to change anything, please be might guest. I am over this one. does not seem like a good investment of time.